src/vs/platform/instantiation/common/instantiationService.ts

```diff
@@ -72,6 +72,9 @@ export class InstantiationService implements IInstantiationService {
 		}
 	}
 
+	createInstance(ctorOrDescriptor: any | SyncDescriptor<any>, ...rest: any[]): any; // Comment out this line to fix type issues
+	createInstance<T>(descriptor: SyncDescriptor0<T>): T;
+	createInstance<Ctor extends new (...args: any[]) => any, R extends InstanceType<Ctor>>(ctor: Ctor, ...args: GetLeadingNonServiceArgs<ConstructorParameters<Ctor>>): R;
 	createInstance(ctorOrDescriptor: any | SyncDescriptor<any>, ...rest: any[]): any {
```

Comment:
The typing seems pretty intense.
Does it narrow down the valid arguments that can be passed to a `createInstance` call? If so, I wonder why only the `TestInstantiationService` was affected?

Reply:
This is copied from https://github.com/microsoft/vscode/blob/559e9beea981b47ffd76d90158ccccafef663324/src/vs/platform/instantiation/common/instantiation.ts#L55

The root cause is that our tests are using a concrete version of `InstantiationService` instead of the interface (`IInstantiationService`). This means they see InstantiationService's signature for `createInstance` instead of seeing the correct signature. This is the expected behavior for TS but happens to cause issues in  this case